certsuite/harness.py

```diff
@@ -421,6 +420,49 @@ def check_device_profile(device_profile_path):
                     logger.error('Invalide device profile file [%s]' % device_profile_path)
     except:
         logger.critical("Encountered error at checking device profile file [%s]:\n%s" % (device_profile_path, traceback.format_exc()))
+    finally:
+        return device_profile_object
```

Comment:
I wonder would that cause any trouble if there's a valid json file without 'result' in it would return to caller.